### DISCUSSION TITLE
Add web_search extension to the official list

### DISCUSSION THREAD

**Discussion by iChristGit** [oobabooga/text-generation-webui#3557] [Ideas]
The extension:

https://github.com/simbake/web_search

Is a great way to get current info or specific wikipedia article summaries just with the prompt "search X wikipedia"
It can do almost everything, give sport scores, get correct time and weather and fill the gaps with local llms.
I found this extension very helpful and would like it to be further recognized and worked on!
What are your thoughts?
This should be an official addon!

**Comment by ShadowStormDragon** [comment]
I agree with you. There are two main things that I would love. A secure connection for the LLM to access the web through this client that would allow it to do searches to help fill the gaps in its knowledge, like Bing, but not be dependent on Open AI, Microsoft, or any compony without a rock solid privacy and security. So, the LLM would be restricted to searching for what is relevant like a certain recent occurrence that would not be in the LLMs database. Otherwise, I don't see the point of an LLM being dependent on a service that one could just log into and use instead of logging into it through this and still compromising ones information. I would want there to be some way to make sure the connection to the web would be promptly closed after the inquiry. The only other alternative, for those who don't have the money for a ton of GPUs to train it would be web scraping that I believe is pretty expensive.

**Comment by ghost** [comment]
Hi all, I am the developer of web_search and am glad for the recognition it gets every day. I am happy to see people like it and will continue to develop it adding extra features. Suggestions are welcomed. I am working on making the information scraped to not overwhelm the model and make sure it spits out the correct info. I am still new to python and the AI stuff and learning as I go along.

Thanks to TheCyberViking for the initial template.

**Comment by Charlotte-br560** [comment]
"Thanks for suggesting the web_search extension! Its versatility is impressive. I support its addition to the official list.

**Reply by TheLounger** [reply]
- No longer maintained
- Relies on Google Chrome browser 🧐

Doesn't sound like a good idea to add this right now.
